tensorrt_llm/llmapi/llm_args.py

```
@@ -166,6 +166,59 @@ def _generate_cuda_graph_batch_sizes(max_batch_size: int,
+        default="rocket", description="The algorithm for sparse attention.")
+
+    @classmethod
+    def from_dict(cls, data: dict):
+        # dispatch to the correct sparse attention config
+        config_classes = {
+            "Rocket": RocketSparseAttentionConfig,
+        }
+
+        config_class = config_classes.get("algorithm")
+        if config_class is None:
+            raise ValueError(f"Invalid algorithm")
```

Ruff (0.13.1)

Avoid specifying long messages outside the exception class

(TRY003)